src/jit/lower.cpp

```diff
@@ -924,6 +924,7 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
             {
                 var_types regType          = fieldListPtr->gtGetOp1()->TypeGet();
                 argSplit->m_regType[index] = regType;
+                fieldListPtr->gtRegNum     = REG_NA;
```

Comment:
I also thought that clearing is needless, but some test failed when I removed clearing in `LowerArg()`. We need more investigation.